examples/create-react-app/src/App.js

```diff
@@ -12,168 +12,152 @@ import {
   ChoiceList,
   SettingToggle,
 } from '@shopify/polaris';
-
-class App extends Component {
-  state = {
-    first: '',
-    last: '',
-    email: '',
-    checkboxes: [],
-    connected: false,
-  };
-
-  render() {
-    const breadcrumbs = [
-      {content: 'Sample apps'},
-      {content: 'Create React App'},
-    ];
-    const primaryAction = {content: 'New product'};
-    const secondaryActions = [{content: 'Import', icon: 'import'}];
-
-    const choiceListItems = [
-      {label: 'I accept the Terms of Service', value: 'false'},
-      {label: 'I consent to receiving emails', value: 'false2'},
-    ];
-
-    const accountMarkup = this.state.connected
-      ? this.disconnectAccountMarkup()
-      : this.connectAccountMarkup();
-
-    return (
-      <Page
-        title="Polaris"
-        breadcrumbs={breadcrumbs}
-        primaryAction={primaryAction}
-        secondaryActions={secondaryActions}
-      >
-        <Layout>
-          <Layout.AnnotatedSection
-            title="Style"
-            description="Customize the style of your checkout"
+import {ImportMinor} from '@shopify/polaris-icons';
+
+export default function App() {
+  const [first, setFirst] = useState('');
+  const [last, setLast] = useState('');
+  const [email, setEmail] = useState('');
+  const [checkboxes, setCheckboxes] = useState([]);
+  const [connected, setConnected] = useState(false);
+
+  const handleFirstChange = useCallback((value) => setFirst(value), []);
```

Comment:
See https://reactjs.org/docs/hooks-reference.html#usecallback.

If we were to define this inline then we pass a new function into the child component on every rerender. This breaks optimization mechanisms like [PureComponent](https://reactjs.org/docs/react-api.html#reactpurecomponent) and [React.memo](https://reactjs.org/docs/react-api.html#reactmemo) because they do a shallow compare of props to see if they can skip rerendering. This is because functions are compared using reference - `(() => {}) == (() => {})` evaluates to false.

By using `useCallback` we ensure that we always pass the same function into child components and thus components using React.memo (like TextField) can skip rerendering and we get performance boosts.